### ISSUE TITLE
Support indentation w/ 2 spaces

### ISSUE THREAD

**Issue by rofinn** [chyh1990/yaml-rust#101]
It appears that yaml-rust only supports 4 spaces for indentation.

**Comment by iamed2** [comment]
http://yaml.org/spec/1.2/spec.html#id2777534

The YAML 1.2 spec says that any number of spaces can be used to represent an indentation level.

**Comment by kbknapp** [comment]
This has come up a few times in [`clap`](https://github.com/kbknapp/clap-rs) with an open issue being kbknapp/clap-rs#965

**Comment by iamed2** [comment]
`clap` was indeed how Rory discovered this

**Comment by hoodie** [comment]
I don't think this is true, as I commented on kbknapp/clap-rs#965 already. I also have some [tests](https://github.com/chyh1990/yaml-rust/pull/104) that should demonstrate this. Or do I not understand correctly?

**Comment by kbknapp** [comment]
@hoodie I agree the specific test referenced in clap#965 doesn't quite line up with this issue, however just today I ran into an issue where the only change made was indents from 4 to 2 and it works with 4 but fails with 2. 

Once I get to a computer I can post the actual test for reference.

**Comment by berwyn** [comment]
Just ran into this today via Clap. Having a background with Rails and Jekyll, I wrote my Clap yaml with two spaces for indentation and then was beat over the head with weird issues where the yaml would fail to parse. Took some digging and found out that indentation by two spaces is what killed me. Given that every other yaml parser I've used hasn't cared about the number of spaces present, but rather the number of spaces relative to the parent, I'd love to see this functionality added!
